pandas/core/tools/datetimes.py

```diff
@@ -638,7 +640,7 @@ def to_datetime(
     infer_datetime_format: bool = ...,
     origin=...,
     cache: bool = ...,
-) -> DatetimeScalar | NaTType:
+) -> Timestamp | NaTType:
```

Comment:
Are scalars not returned by this function?

The previous version maintained sub-classes of Timestamp (and I think also used the same Scalar dtype):
```py
@overload
def to_datetime(
    arg: DatetimeScalar,  # this was a TypeVar
    ...
) -> DatetimeScalar | NaTType:
    ...
```

Reply:
This is for an override.  In this file `Scalar` is defined as `Union[int, float, str]` and `DatetimeScalar` is defined as `Union[Scalar, datetime]`.  In all cases, when those types are passed, the result is a `Timestamp`: 
```python
>>> import datetime
>>> pd.to_datetime(datetime.datetime(year=2022,month=2,day=22))
Timestamp('2022-02-22 00:00:00')
>>> pd.to_datetime(12345678)
Timestamp('1970-01-01 00:00:00.012345678')
>>> pd.to_datetime(12345678.123497812)
Timestamp('1970-01-01 00:00:00.012345678')
>>> pd.to_datetime("2022-01-03")
Timestamp('2022-01-03 00:00:00')
```

Reply:
Okay, thanks for the explanation!